scripts/extract-paired-reads.py

```diff
@@ -98,45 +71,21 @@ def main():
     print >>sys.stderr, 'outputting interleaved pairs to "%s.pe"' % outfile
     print >>sys.stderr, 'outputting orphans to "%s.se"' % outfile
 
-    last_record = None
-    last_name = None
-
     n_pe = 0
     n_se = 0
 
-    record = None
-    index = 0
-    for index, record in enumerate(screed.open(sys.argv[1])):
+    screed_iter = screed.open(args.infile)
+    for index, is_pair, read1, read2 in broken_paired_reader(screed_iter):
         if index % 100000 == 0 and index > 0:
-            print '...', index
-        name = record['name'].split()[0]
-
-        if last_record:
-            if is_pair(last_name, name):
-                paired_fp.write(output_pair(last_record, record))
-                name, record = None, None
-                n_pe += 1
-            else:
-                single_fp.write(output_single(last_record))
-                n_se += 1
-
-        last_name = name
-        last_record = record
-
-    if last_record:
-        if is_pair(last_name, name):
-            paired_fp.write(output_pair(last_record, record))
-            name, record = None, None
+            print >>sys.stderr, '...', index
```

Comment:
Maybe?

But a few thoughts --

this isn't diagnostic output;
if there's one thing that's well tested "in the wild", it's this;
the acceptance tests certainly cover this line.

So I personally think this is simply not worth the extra effort and code.